File: arrays 101/valid_mountain_array.java
class Solution {
    public boolean validMountainArray(int[] A) {
        int i, index=0;
        int n = A.length;
        
        if(n==0)
        {
            return false;
        }
        
        boolean flag1=false, flag2=false;
        
        if(n==0)
        {
            return false;
        }
        for(i=1; i<n; i++)
        {
            if(A[i] > A[i-1])
            {
                index = i;
            }
        }
        
        for(i=0; i<index; i++)
        {
            if(A[i] < A[i+1])
            {
               flag1 = true;
            }
            
            else{
                flag1 = false;
                break;
            }
        }
        
        for(i=index + 1; i<n; i++)
        {
            if(A[i] < A[i-1])
            {
               flag2 = true;
            }
            
            else{
                flag2 = false;
                break;
            }
        }
        if(flag1==true && flag2==true)
        {
            return true;
        }
        else{
            return false;
        }
        
    }
}

File: arrays 101/valid_mountain_array.py
class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        
        if len(A)==0: 
            return False
        mid = 0
        for i in range(1, len(A)):
            if A[i] > A[i-1]:
                mid = i
        flag1 = False
        for i in range(0,mid):
            if A[i]<A[i+1]:
                flag1=True
            else:
                flag1=False
                break
        flag2 = False
        for i in range(mid+1,len(A)):
            if A[i-1]>A[i]:
                flag2=True
            else:
                flag2=False
                break
         
        if flag1== True and flag2==True:
            return True
        else:
            return False